File: tyvj/第05页(1400-1499)/1474打鼹鼠/1474.pas
type
  rec=record x,y,k:longint;end;
var
  a:array[1..10000] of rec;
  i,j,m,n,x,y,x1,y1,ans,k:longint;
function ok(p:rec):boolean;
begin
  if (p.x>x1)or(p.x<x) then exit(false);
  if (p.y>y1)or(p.y<y) then exit(false);
  exit(true);
end;
begin
  read(n);
  i:=0;
  while true do
    begin
      read(m);
      if m=3 then break;
      if m=1 then begin read(x,y,k);inc(i);a[i].x:=x;a[i].y:=y;a[i].k:=k;end;
      if m=2 then
        begin
          read(x,y,x1,y1);
          ans:=0;
          for j:=1 to i do
            if ok(a[j]) then inc(ans,a[j].k);
          writeln(ans);
        end;
    end;
end.

File: tyvj/第05页(1400-1499)/1474打鼹鼠/1474.cpp
#include <cstdio>
struct rec{int x,y,k;};
rec a[10010];
int n,m,x1,y1,x2,y2,x,y,k,p,ans;
int main(){
	scanf("%d",&n);
	while (1){
		scanf("%d",&m);
		if (m == 3) break;
		if (m == 1){
			scanf("%d%d%d",&x,&y,&k);
			p++;
			a[p].x = x;
			a[p].y = y;
			a[p].k = k;
		}
		if (m == 2){
			ans = 0;
			scanf("%d%d%d%d",&x1,&y1,&x2,&y2);
			for (int i = 1;i <= p;i++) if (x1 <= a[i].x && a[i].x <= x2 && y1 <= a[i].y && a[i].y <= y2) ans += a[i].k;
			printf("%d\n",ans);
		}
	}
}
